codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/EndpointGenerator.java

```diff
@@ -387,6 +396,12 @@ private void generateInternalResolverImplementation(GoWriter writer) {
         writer.write("");
         writer.writeDocs("ResolveEndpoint resolves the service endpoint for the given region and options");
         writeInternalResolveEndpointImplementation(writer, resolverImplSymbol, "r", () -> {
+            // Currently all APIs require a region to derive the endpoint for that API. If there are ever a truly
+            // region-less API then this should be gated at codegen.
+            writer.addUseImports(AwsGoDependency.AWS_CORE);
```

Comment:
Any reason for doing that at this level rather then https://github.com/aws/aws-sdk-go-v2/blob/master/internal/endpoints/endpoints.go ?

Reply:
Primary reason to do it per service so that it would be easier for a service to opt out of this behavior if ever needed. With the idea that by default all API clients require a region.